### PR TITLE
Improve Refinement in cegar values

### PR DESCRIPTION
**Pull request by ahmed-irfan**
This PR changes how the refinement in cegar-values is done. Essentially, it utilizes the refine-ts method from ceg-prophecy.

### CODE REVIEW

File: engines/cegar_values.cpp [hunk 1]
```diff
@@ -317,37 +317,37 @@ bool CegarValues<Prover_T>::cegar_refine()
...
 template <>
 void CegarValues<CegProphecyArrays<IC3IA>>::refine_subprover_ts(
-    const Term & constraint)
+    const UnorderedTermSet & axioms)
 {
-  Term transferred_constraint =
-      from_cegval_solver_.transfer_term(constraint, BOOL);
-  assert(super::abs_ts_.only_curr(transferred_constraint));
-  super::abs_ts_.add_constraint(transferred_constraint);
+  super::refine_ts(axioms);
```
Comment (makaimann) [inline engines/cegar_values.cpp:350]: Will this only work for another CEGAR loop? E.g. what happens in the future if we allow running `CegarValues` on something other than `CegProphecyArrays<IC3IA>`?

**Review by makaimann — APPROVED** [review]
Looks good to me! I just left a comment to think about for the future.

File: engines/cegar_values.cpp (hunk 1 above)
Comment (ahmed-irfan) [reply engines/cegar_values.cpp:350]: It should also work for other configurations of cegprophecy, e.g. CegProphecyArrays<Interpolantmc>

Comment (makaimann) [reply engines/cegar_values.cpp:350]: Got it, but we'll have to adapt it if we wanted to run it over `k-induction` or `IC3IA` directly, right?

Comment (makaimann) [reply engines/cegar_values.cpp:350]: Which is fine btw, we already would have to adapt it even before this change.

Comment (ahmed-irfan) [reply engines/cegar_values.cpp:350]: Right

**Pull request merged**